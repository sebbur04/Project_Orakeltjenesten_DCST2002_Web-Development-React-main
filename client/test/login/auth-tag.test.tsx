// Import neccessary modules for testing
import * as React from "react";
import { shallow } from "enzyme";
import TagComp from "../../src/login/tag";

// Mock services
jest.mock("../../src/service/tag-service", () => ({
  getTag: jest.fn().mockResolvedValue({ id: 1, name: "Test Tag" }),
  updateTag: jest.fn().mockResolvedValue({}),
  deleteTag: jest.fn().mockResolvedValue({}),
  getPageFromTag: jest.fn().mockResolvedValue([{ id: 1, name: "Page 1" }]),
}));

jest.mock("../../src/service/user-service", () => ({
  getAuthenticatedUser: jest.fn().mockResolvedValue({
    id: 1,
    username: "testuser",
    avatar: "",
    bio: "",
    permid: 1,
  }),
  getAllUsers: jest
    .fn()
    .mockResolvedValue([
      { id: 1, username: "testuser", avatar: "", bio: "", permid: 1 },
    ]),
}));

jest.mock("../../src/service/permission-service", () => ({
  getPermission: jest.fn().mockResolvedValue({ id: 1, tags: true }),
}));

jest.mock("../../src/login/sidebar", () => ({
  instance: () => ({
    mounted: jest.fn(),
  }),
}));

// Test cases
describe("TagComp component tests", () => {
  test("TagComp matches snapshot", (done) => {
    const wrapper = shallow(<TagComp match={{ params: { id: 1 } }} />);
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
