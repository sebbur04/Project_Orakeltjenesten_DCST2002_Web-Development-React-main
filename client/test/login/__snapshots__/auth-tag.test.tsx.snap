// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagComp component tests TagComp matches snapshot 1`] = `
<div
  className="mainContent"
>
  <Card
    title=""
  >
    <div
      className="tag-box-style"
    >
      <Row>
        <div>
          <Row>
            <Column
              width={4}
            >
              <h3 />
            </Column>
            <Column
              width={4}
            >
              <ButtonLight
                onClick={[Function]}
              >
                Endre tag
              </ButtonLight>
            </Column>
          </Row>
        </div>
      </Row>
      <Row />
      <Row>
        <Column>
          Antall sider: 0
        </Column>
      </Row>
    </div>
  </Card>
</div>
`;
